charts/cactus-backend/templates/configmap.yaml

```
@@ -0,0 +1,13 @@
+{{- if .Values.configFiles.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "cactus-backend.fullname" . }}-config
+  labels:
+    {{- include "cactus-backend.labels" . | nindent 4 }}
+data:
+  {{- range $filename, $content := .Values.configFiles.files }}
+  {{ $filename }}: |
+    {{- $content | nindent 4 }}
+  {{- end }}
```

I think it can be simplified like this:
```suggestion
  {{- toJson .Values.configFiles.files }}
```

---

now it looks like
```
---
# Source: cactus-backend/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: release-name-config-files
  labels:
    helm.sh/chart: cactus-backend-0.5.6
    app.kubernetes.io/name: cactus-backend
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.2"
    app.kubernetes.io/managed-by: Helm
data:
  domain_external_mappings.json: |
    {
      "domain_id": "store_id"
    }
```
it becomes below when using `toJson`
```
# Source: cactus-backend/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: release-name-config-files
  labels:
    helm.sh/chart: cactus-backend-0.5.6
    app.kubernetes.io/name: cactus-backend
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.2"
    app.kubernetes.io/managed-by: Helm
data:{"domain_external_mappings.json":"{\n  \"domain_id\": \"store_id\"\n}\n"}
```

---

Oh, maybe you need to remove the hyphen from the opening marker `{`

---

the data is a list of files, and each file might be a json. configmap data doesn't look right with `toJson`.
```
data:
  {{ toJson .Values.configFiles.files }}
  
 
# Source: cactus-backend/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: release-name-config-files
  labels:
    helm.sh/chart: cactus-backend-0.5.6
    app.kubernetes.io/name: cactus-backend
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.2"
    app.kubernetes.io/managed-by: Helm
data:
  {"domain_external_mappings.json":"{\n  \"domain_id\": \"store_id\"\n}\n"}
  ```

---

`toYaml`? :)

---

ah, yes, `{{- toYaml .Values.configFiles.files | nindent 2 }}` works
```
# Source: cactus-backend/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: release-name-config-files
  labels:
    helm.sh/chart: cactus-backend-0.5.6
    app.kubernetes.io/name: cactus-backend
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "0.2"
    app.kubernetes.io/managed-by: Helm
data:
  domain_external_mappings.json: |
    {
      "domain_id": "store_id"
    }
    ```